.circleci/config.yml

```diff
@@ -244,31 +222,17 @@ caffe2_linux_build_defaults: &caffe2_linux_build_defaults
 
         ((echo "source ./workspace/env" && echo 'sudo chown -R jenkins workspace && cd workspace && ./ci_build_script.sh') | docker exec -u jenkins -i "$id" bash) 2>&1 | ts
 
-        mkdir -p /home/circleci/project/caffe2-ci-env
-        touch /home/circleci/project/caffe2-ci-env/.tmp  # Dummy file so that the persist_to_workspace step won't complain about not finding any file
-
+        # Push intermediate Docker image for next phase to use
         if [ -z "${BUILD_ONLY}" ]; then
           export COMMIT_DOCKER_IMAGE=${DOCKER_IMAGE}-${CIRCLE_SHA1}
           docker commit "$id" ${COMMIT_DOCKER_IMAGE}
           docker push ${COMMIT_DOCKER_IMAGE}
-          echo "declare -x COMMIT_DOCKER_IMAGE=${COMMIT_DOCKER_IMAGE}" > /home/circleci/project/caffe2-ci-env/COMMIT_DOCKER_IMAGE
```

Comment:
If we want to allow retrying a test job, we can't persist things to the intermediate workspace anymore, because CircleCI's retry API doesn't support attaching a workspace :(